go/adbc/driver/bigquery/statement.go

```diff
@@ -0,0 +1,525 @@
+				TypeKind: "NULL",
+			},
+			Value: "NULL",
+		}
+		return parameter
+	}
+
+	// https://cloud.google.com/bigquery/docs/reference/storage#arrow_schema_details
+	// https://cloud.google.com/bigquery/docs/reference/rest/v2/StandardSqlDataType#typekind
+	switch value.DataType().ID() {
+	case arrow.BOOL:
+		// GoogleSQL type: BOOLEAN
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "BOOL",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.INT8, arrow.INT16, arrow.INT32, arrow.INT64, arrow.UINT8, arrow.UINT16, arrow.UINT32, arrow.UINT64:
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "INT64",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.FLOAT32, arrow.FLOAT64:
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "FLOAT64",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.BINARY, arrow.BINARY_VIEW, arrow.LARGE_BINARY:
+		// todo: Encoded as a base64 string per RFC 4648, section 4.
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "BYTES",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.STRING, arrow.STRING_VIEW, arrow.LARGE_STRING:
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "STRING",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.DATE32, arrow.DATE64:
+		// todo: Encoded as RFC 3339 full-date format string: 1985-04-12
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "DATE",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.TIMESTAMP:
+		// todo: Encoded as an RFC 3339 timestamp with mandatory "Z" time zone string: 1985-04-12T23:20:50.52Z
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "TIMESTAMP",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.TIME32, arrow.TIME64:
+		// todo: Encoded as RFC 3339 partial-time format string: 23:20:50.52
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "TIME",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.DECIMAL128:
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "NUMERIC",
+			},
+			Value: value.ValueStr(0),
+		}
+	case arrow.DECIMAL256:
+		parameter.Value = &bigquery.QueryParameterValue{
+			Type: bigquery.StandardSQLDataType{
+				TypeKind: "BIGNUMERIC",
+			},
+			Value: value.ValueStr(0),
+		}
+	}
+
+	return parameter
+}
+
+func (st *statement) getQueryParameter() ([]bigquery.QueryParameter, error) {
+	var values arrow.Record
+	if st.paramBinding != nil {
+		values = st.paramBinding
+	} else if st.streamBinding != nil {
+		if st.streamBinding.Next() {
+			values = st.streamBinding.Record()
+		} else {
```

Comment:
in the case of an ExecuteUpdate, should we be looping over the stream and execute until we have no more?